tsconfig.json

```diff
@@ -17,7 +17,8 @@
         "noUnusedLocals": true,
         "lib": ["dom", "es6", "esnext.asynciterable"],
         "skipLibCheck": true,
-        "jsx": "preserve"
+        "jsx": "preserve",
+        "experimentalDecorators": true
```

Comment:
👀

Reply:
Yeah I need to remove some of the test/experimental stuff from this PR still
Can always come back to decorators later (if ever)